### ISSUE TITLE
[ENH] Additional strategy to input forecasted exogenous features to forecasters

### ISSUE THREAD

**Issue by benHeid** [sktime/sktime#5620]
**Is your feature request related to a problem? Please describe.**
Currently the behavior in sktime when using exogenous features is that the exogenous features are considered as a time series and according to the time point that is considered right now in training or forecasting the corresponding element is taken out of the time series. 

However, if we have forecasts as input this solution has its limitation. Consider the following scenario: 
* We want to do forecasting of the electricity demand based on weather forecasts.  
* We have historical demand time series but also historical forecasts. Historical forecasts means that we have different forecasts of the next n value for each point in time. In other words, the forecasts axis are `creation time of the forecast` and `forecast horizon`. 

I.e., consider during training the time point t and t + 1.  At time `t` the training sample consists as ground truth the next `n` values but only the forecasted exogenous features created at time `t` should be used. 
Note, this might get even more complicated, if we have forecasted exogenous features and some dummy exogenous features as calendar information. The calendar information could be passed as in the current behavior, but the forecasted exogenous features should be handled differently. 

I am unsure if all algorithm could handle such cases. However, at least the reducers could handle theoretically such exogenous forecasts in the desired way and all neural networks. 

I am aware that this issue is probably written very confusing. So feel free to ask for clarifications.

**Comment by fkiraly** [comment]
I think I understand, kindly check if I do:

There are two parts of the statement, one about forecasts at different horizon, an one about how to use them in exogenous forecast related strategies.

### forecasts at different horizons

if you have a forecaster, you can run it in update/predict mode. So for each cutoff $c$ and horizon $h$, you can produce a value $\tilde{y}_{c, h}$, which is a forecast value for the value time stamp $c+h$.

If you have equally spaced time stamps, then for a time stamp $t$ you can have multiple forecasts, namely potentially all $\widehat{y}_{t, h} := \tilde{y}_{t-h, h}$, with $h<t$.

In particular, there are potentially multiple forecasts for a "true" value $y_t$ at time stamp $t$, along the above lines, parameterized alternatively by cutoff (from which knowledge state the forecast was made), or by horizon (how far ahead). This leaves aside randomness, or that forecasts made could be influenced by history of updates, etc.

### how to use these in exogenous forecasts

This is where my understanding what you want to do/ask gets patchy, I think.

We can in-principle produce these forecasts, and use them in sth like a reducer, in various ways.

But, what do you mean by "if all algorithms could handle such cases"? In general, we only input known, not forecasted values through the generic interface points. Most algorithms do not compute auxiliary exogenous forecasts internally.

**Comment by fkiraly** [comment]
... don't get why my tex does not properly render. Looks like a bug in the TeX/GitHub integration, where do we report that?

Seems it prioritizes _underscore_ formatting above TeX formula delimiters, it should be the other way round.

**Comment by benHeid** [comment]
I this is exactly, what I meant. 

Regarding How to use these in exogenous forecasts:

I think the first is that I have them as exogenous forecasts that are part of the provided data. So I have not to create them using ForecastX or something similar. Now the question is, how would you pass these forecasted data into a a forecaster. Possible side constraints are:
* Some of the exogenous data are forecasts, some not.
* The frequency of the forecasts of the exogenous data differs from its resolution and the resolution of the forecast I want to create with sktime. E.g., weather data, often the frequency of the creation of weather forecasts is six hours, however, each forecast itself has an hourly resolution. 

Regarding if all algorithms could handle these cases, I have to explain my desired solution, which is focussing on neural networks, or reducers. 
For neural networks you are creating for each point in time a new sample that should be forecasted. So here, I would like to have, that based on the provided exogenous features, the correct forecast would be used. I make this statement on not sure if each algorithm could handle these more because I am directly knowing that for reducers and neural networks this should be easily implementable as far we are decided how we would like to pass these information.

**Comment by fkiraly** [comment]
Summarizing discussion with @benHeid today, starting with an example:

* suppose your *exogenous data* are historical weather forecasts. These come with cutoff and horizon both, e.g., the 1-day ahead forecast from Dec 5, or the 2-day ahead forecast from Dec 4, both being a forecast of the weather on Dec 6 (each a single value for simplicity)
* When considering exogenous data in the `predict` phase, in general only some of these will be available at a given time stamp,
    * let's say, we put all the forecasts "for" a given day in the same row. Then, if current time is Dec 5, Dec 6 has an 1- and 2-day forecast available, but Dec 7 only a 2-day forecast - because the 1-day forecast will be made on Dec 6, which has not happened yet.
* now suppose, you want to use the "for" forecasts in training or evaluating a model. Then, the logic of this needs to know somehow which data of a complete set of 1- and 2-day forecasts are available when

If I understand the question well, it is now "how should this look like" - and is an API change helpful?

Some notes:

   * passing information about how data is revealed could be achieved by putting data with the same cutoff always i the same row of exogenous data, e.g., by the "from" date instead of "of" date, i.e., all forecasts made on Dec 5, then all on Dec 6, etc, in one row
    * another option is the darts interface, where variables known in the past are distinguished from variables known in the future. But this does not allow intermediate graduations, e.g, known until 2 days from know (like weather forecast data)
